src/screens/DiscoveryScreenV2/fragment/NewsFragment.js

```diff
@@ -1,140 +1,161 @@
-    if (news.length === 0 && !isFirstTimeOpen) return <View style={styles.noDataFoundContainer}>
+    return news.map((item, index) => {
+      const contentParam = newsDiscoveryContentParamBuilder(
+        item.title,
+        item.image,
+        item.description,
+        item.news_url,
+        item?.newsLinkDomain?.logo,
+        item?.newsLinkDomain?.domain_name || item?.site_name,
+        item.createdAt,
+        item.news_link_id,
+        item?.newsLinkDomain?.credder_score,
+        item?.post_id
+      );
+
+      const onContentClicked = () => {
+        navigation.navigate('DetailDomainScreen', {
+          item: {
+            id: item.post_id,
+            score: item?.newsLinkDomain?.credder_score
+          }
+        });
+      };
+
+      return (
+        <RenderItem
+          key={`newsDiscovery-${index}`}
+          selfUserId={myId}
+          item={contentParam}
+          onPressShare={share.shareNews}
+          // onPressContent={onContentClicked}
+        />
+      );
+    });
+  };
+
+  if (!isReady) return <></>;
+
+  if (isLoadingDiscoveryNews)
+    return (
+      <View style={styles.fragmentContainer}>
+        <LoadingWithoutModal />
+      </View>
+    );
+  if (news.length === 0 && !isFirstTimeOpen)
+    return (
+      <View style={styles.noDataFoundContainer}>
         <Text style={styles.noDataFoundText}>No news found</Text>
+      </View>
+    );
+
+  return (
+    <View>
+      <RecentSearch
+        shown={isFirstTimeOpen}
+        setSearchText={setSearchText}
+        setIsFirstTimeOpen={setIsFirstTimeOpen}
+      />
+      {renderNewsItem()}
+      <View style={styles.padding} />
     </View>
-
-    return <View>
-        <RecentSearch shown={isFirstTimeOpen}
-            setSearchText={setSearchText}
-            setIsFirstTimeOpen={setIsFirstTimeOpen} />
-        {renderNewsItem()}
-        <View style={styles.padding} />
-    </View>
-}
+  );
+};
 
 const styles = StyleSheet.create({
-    fragmentContainer: {
-        flex: 1,
-        backgroundColor: colors.concrete,
-    },
-    noDataFoundContainer: {
-        flex: 1,
-        backgroundColor: colors.white,
-        justifyContent: 'center',
-    },
-    noDataFoundText: {
-        alignSelf: 'center',
-        justifyContent: 'center',
-        fontFamily: fonts.inter[600],
-    },
-    padding: {
-        height: 16,
-    },
-    unfollowedHeaders: {
-        fontFamily: fonts.inter[600],
-        marginLeft: 20,
-    }
-})
-
-export default NewsFragment
\ No newline at end of file
+  fragmentContainer: {
+    flex: 1,
+    backgroundColor: COLORS.concrete
+  },
+  noDataFoundContainer: {
+    flex: 1,
+    backgroundColor: COLORS.white,
+    justifyContent: 'center'
+  },
+  noDataFoundText: {
+    alignSelf: 'center',
+    justifyContent: 'center',
+    fontFamily: fonts.inter[600]
+  },
+  padding: {
+    height: 16
+  },
+  unfollowedHeaders: {
+    fontFamily: fonts.inter[600],
+    marginLeft: 20
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The color values have been updated to use the `COLORS` constants. This is a good practice as it ensures consistency in color usage across the application. Make sure that the new color values are consistent with the design specifications.

Overall, the changes look good and improve the maintainability and consistency of the code. The only potential issue is the commented-out `onPressContent` prop in the `RenderItem` components, which should be addressed based on whether this functionality is intended to be disabled or not.


<!-- This is an auto-generated comment by OSS CodeRabbit -->